Zend/zend_vm_def.h

```diff
@@ -9808,6 +9808,48 @@ ZEND_VM_HANDLER(209, ZEND_INIT_PARENT_PROPERTY_HOOK_CALL, CONST, UNUSED|NUM, NUM
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HANDLER(210, ZEND_FETCH_DEFAULT_ARG, UNUSED|NUM, UNUSED)
+{
+	USE_OPLINE
+	SAVE_OPLINE();
+
+	// Initialize result.
+	ZVAL_UNDEF(EX_VAR(opline->result.var));
+
+	zend_function *called_func = EX(call)->func;
+
+	reflection_parameter_reference param;
+	param.offset = opline->op1.num - 1;
+	param.required = param.offset < called_func->common.required_num_args;
+	param.fptr = called_func;
+
+	if (UNEXPECTED(param.required || param.offset >= called_func->common.num_args)) {
+		zend_value_error("Cannot pass default to %s parameter %u of %s%s%s()",
+			param.required ? "required" : "undeclared",
+			opline->op1.num,
+			called_func->common.scope ? ZSTR_VAL(called_func->common.scope->name) : "",
+			called_func->common.scope ? "::" : "",
+			ZSTR_VAL(called_func->common.function_name)
+		);
+		HANDLE_EXCEPTION();
+	}
+
+	param.arg_info = &called_func->common.arg_info[param.offset];
+
+	zval default_value;
+	reflection_get_parameter_default(&default_value, &param);
```

Comment:
As mentioned off-list, you need to handle the failure path here.

Reply:
- How can we simulate a failure here? 
- Considering C doesn't have try/catch, how can we handle it?